In [2]:
import re
import zipfile
import os
import sys
import lucem_illud_2020
import pandas as pd

In [2]:
corpus_name = "/home/rkcatipon/Content-Analysis-2020/NOW_copy"


In [3]:
def loadcorpus(corpus_name, corpus_style="text"):
    texts_raw = {}
    for file in os.listdir(corpus_name + "/"):
        if corpus_style in file:
            print(file)
            zfile = zipfile.ZipFile(corpus_name + "/" + file)
            for file in zfile.namelist():
                texts_raw[file] = []
                with zfile.open(file) as f:
                    for line in f:
                        texts_raw[file].append(line)
    return texts_raw

In [4]:
news_raw = loadcorpus(corpus_name)

text-16-12.zip
text-16-11.zip


In [5]:
def clean_raw_text(raw_texts):
    clean_texts = []
    for text in raw_texts:
        try:
            text = text.decode("utf-8")
            clean_text = text.replace(" \'m", "'m").replace(" \'ll", "'ll").replace(" \'re", "'re").replace(" \'s", "'s").replace(" \'re", "'re").replace(" n\'t", "n't").replace(" \'ve", "'ve").replace(" /'d", "'d")
            clean_texts.append(clean_text)
        except AttributeError:
            # print("ERROR CLEANING")
            # print(text)
            continue
        except UnicodeDecodeError:
            # print("Unicode Error, Skip")
            continue
    return clean_texts

In [8]:
zfile = zipfile.ZipFile(corpus_name + "/sources-16-11.zip")
source = []

In [9]:
for file in zfile.namelist():
    with zfile.open(file) as f:
        for line in f:
            source.append(line)

In [10]:
# adding secdond source
zfile2 = zipfile.ZipFile(corpus_name + "/sources-16-12.zip")
for file in zfile2.namelist():
    with zfile2.open(file) as f:
        for line in f:
            source.append(line)

In [11]:
source[0:2]

[b'15152901\t902\t16-11-01\tUS\tThe Jewish Press\thttp://www.jewishpress.com/multimedia/video-picks/ancient-arabic-inscription-identifies-the-dome-of-the-rock-with-the-jewish-temple-video/2016/11/01/\tAncient Arabic Inscription Identifies the Dome of the Rock with the ...\r\n',
 b'15153736\t2978\t16-11-01\tUS\tPOWER magazine\thttp://www.powermag.com/state-solar-new-tech-outdated-rate-designs/\tThe State of Solar: New Tech, Outdated Rate Designs\r\n']

In [16]:
news_texts = {}
just_texts = {}
for files in news_raw :
    news = clean_raw_text(news_raw[files][1:])
    for article in news:
        txts = lucem_illud_2020.word_tokenize(article)
        try:
            news_texts[txts[0][2:]] = txts[1:]
            just_texts[txts[0][2:]] = article
        except IndexError:
            continue
            
 

KeyboardInterrupt: 

In [13]:
news_texts.keys()

{'': []}

In [ ]:
news_df = pd.DataFrame(columns=["title", "date", "country", "source", "url", "text", "tokenized texts"])

In [ ]:
for news in source[3:]:
    try:
        fileid, total_words, date, country, source, url, title = news.decode("utf-8").split("\t")
    except UnicodeDecodeError:
        continue
    try:
        news_df.loc[fileid.strip()] = [title.strip(), date.strip(), country.strip(), source.strip(), url.strip(), just_texts[fieldid.strip()], news_texts[fileid.strip()]]
    except KeyError:
        continue

In [ ]:
news_df.head()

[b'15152901\t902\t16-11-01\tUS\tThe Jewish Press\thttp://www.jewishpress.com/multimedia/video-picks/ancient-arabic-inscription-identifies-the-dome-of-the-rock-with-the-jewish-temple-video/2016/11/01/\tAncient Arabic Inscription Identifies the Dome of the Rock with the ...\r\n',